test/parallel/xyk-pallet.multiswap2hops.test.ts

```diff
@@ -139,46 +137,44 @@ describe("Multiswap [2 hops] - happy paths", () => {
     const listIncludingSmallPool = tokenIds.concat([assetIdWithSmallPool]);
     testUser1.addAssets(listIncludingSmallPool);
     await testUser1.refreshAmounts(AssetWallet.BEFORE);
-    const multiSwapOutput = await multiSwapSell(
+    
+    const multiswapSellPaymentInfo = await getMultiswapSellPaymentInfo(
       testUser1,
       listIncludingSmallPool,
       swapAmount,
       BN_TEN_THOUSAND,
     );
-    const eventResponse = getEventResultFromMangataTx(multiSwapOutput, [
-      "xyk",
-      "MultiSwapAssetFailedOnAtomicSwap",
-    ]);
+    
+    const multiSwapOutput = await multiSwapSellMarket(
+      testUser1,
+      listIncludingSmallPool,
+      swapAmount,
+      BN_TEN_THOUSAND,
+    );
+    const eventResponse = getEventResultFromMangataTx(multiSwapOutput);
+    expect(eventResponse.state).toEqual(ExtrinsicResult.ExtrinsicFailed);
+    expect(eventResponse.data).toEqual("InsufficientOutputAmount");
     await testUser1.refreshAmounts(AssetWallet.AFTER);
     const walletsModifiedInSwap = testUser1.getWalletDifferences();
-    //Validate that the modified tokens are MGX and the first element in the list.
-    expect(walletsModifiedInSwap).toHaveLength(2);
+    //Validate that the modified tokens are only GASP in the list.
+    expect(walletsModifiedInSwap).toHaveLength(1);
     expect(
       walletsModifiedInSwap.some((token) => token.currencyId.eq(GASP_ASSET_ID)),
     ).toBeTruthy();
-    expect(
-      walletsModifiedInSwap.some((token) =>
-        token.currencyId.eq(listIncludingSmallPool[0]),
-      ),
-    ).toBeTruthy();
+    // expect(
+    //   walletsModifiedInSwap.some((token) =>
+    //     token.currencyId.eq(listIncludingSmallPool[0]),
+    //   ),
+    // ).toBeTruthy();
     const changeInSoldAsset = walletsModifiedInSwap.find((token) =>
-      token.currencyId.eq(listIncludingSmallPool[0]),
+      token.currencyId.eq(GASP_ASSET_ID),
     )?.diff.free;
-    const expectedFeeCharged = swapAmount
-      .muln(3)
-      .divn(1000)
-      .add(new BN(3))
-      .neg();
-    expect(changeInSoldAsset).bnEqual(expectedFeeCharged);
-    expect(eventResponse.state).toEqual(ExtrinsicResult.ExtrinsicSuccess);
-    //check only 0.3%
-    expect(
-      multiSwapOutput.findIndex(
-        (x) =>
-          x.section === EVENT_SECTION_PAYMENT ||
-          x.method === EVENT_METHOD_PAYMENT,
-      ),
-    ).toEqual(-1);
+    expect(changeInSoldAsset).bnEqual(multiswapSellPaymentInfo.neg());
+    //pay transaction fee?
```

Comment:
In new pallet user pay transferFees for this extrinsic, right?

Reply:
yup, if tx fail, user pay in gasp